app/resources/api/v2/feed_resource.rb

```diff
@@ -53,7 +56,10 @@ def self.get_results_from_feed_teams(team_ids, feed_id, query, type, after, webh
         RequestStore.store[:pause_database_connection] = true # Release database connection during Bot::Alegre.request_api
         RequestStore.store[:smooch_bot_settings] = feed.get_smooch_bot_settings.to_h
         results = Bot::Smooch.search_for_similar_published_fact_checks(type, query, feed.team_ids, after, feed_id)
-        Feed.delay({ retry: 0, queue: 'feed' }).save_request(feed_id, type, query, webhook_url, results.to_a.map(&:id)) unless skip_save_request
+        unless skip_save_request || RequestStore.store[:already_saved_request]
+          Feed.delay({ retry: 0, queue: 'feed' }).save_request(feed_id, type, query, webhook_url, results.to_a.map(&:id))
+          RequestStore.store[:already_saved_request] = true
```

Comment:
You added this because you did actually see this method being called more than once for the same request? Or just to be safe?

Reply:
Yes. After adding the skip_save_request filter, sometimes this method is being called more than once.